packages/jest-jasmine2/src/__tests__/jasmine-uses-jests-extend-api-test.js

```diff
@@ -0,0 +1,33 @@
+describe('addMatcher Adapter', () => {
+  beforeEach(() => {
+    jasmine.addMatchers({
+      _toBeValue(util, customEqualityTesters) {
+        return {
+          compare(actual, expected) {
+            const pass = actual == expected;
+
+            return {
+              message: `Expected ${pass} to be same value as ${expected}`,
+              pass,
+            };
+          }
+        }
+      }
+    });
+  });
+
+  it('jasmine.addMatcher calls `expect.extend`', () => {
+    const originalExtend = expect.extend;
+    expect.extend = jest.genMockFunction();
+
+    jasmine.addMatchers({});
+
+    expect(expect.extend).toBeCalled();
```

Comment:
if this test fails by throwing, we won't reset `expect.extend` and we'll be in an odd state. Can you do `afterAll(() => expect.extend = originalExtend)`?